dvc/scm/git/__init__.py

```diff
@@ -124,19 +124,16 @@ def ignore(self, path, in_curr_dir=False):
         )
         logger.info(msg)
 
-        self._add_entry_to_gitignore(entry, gitignore, ignore_list)
+        self._add_entry_to_gitignore(entry, gitignore)
 
         self.track_file(os.path.relpath(gitignore))
 
         self.ignored_paths.append(path)
 
     @staticmethod
-    def _add_entry_to_gitignore(entry, gitignore, ignore_list):
-        content = entry
-        if ignore_list:
-            content = "\n" + content
```

Comment:
And what if there is no newline to start with? Will the current implementation just write
```
oldnew
```
?

Reply:
Yes, fixed it and wrote test for such situation